packages/istanbul-lib-instrument/src/instrumenter.js

```diff
@@ -80,7 +80,13 @@ class Instrumenter {
         const opts = this.opts;
         const ast = babylon.parse(code, {
             allowReturnOutsideFunction: opts.autoWrap,
-            sourceType: opts.esModules ? "module" : "script"
+            sourceType: opts.esModules ? "module" : "script",
+            plugins: [
+              'asyncGenerators',
+              'dynamicImport',
+              'flow',
+              'jsx'
+            ]
```

Comment:
Babylon plugins aren't the same as Babel plugins.

Babylon "plugins" just let you parse more stuff (they aren't separately modules/packages; they are included in babylon (doesn't have any dependencies) and are not on by default since non of the plugins are in the Javascript spec (until Stage 4)

and yes `*` is a bad idea because we may update the experimental proposals since they will change over time.

If you want to be able to parse everything then you can add all the plugins in the list. The estree plugin will back babylon parse with https://github.com/estree/estree